apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx

```diff
@@ -220,11 +220,13 @@ export default function SupabaseListener({ accessToken }) {
   const router = useRouter()
 
   useEffect(() => {
-    supabase.auth.onAuthStateChange((event, session) => {
+    const { data: subscription } = supabase.auth.onAuthStateChange((event, session) => {
       if (session?.access_token !== accessToken) {
         router.refresh()
       }
     })
+  
+    return () => subscription.unsubscribe();
```

Comment:
Thanks for making that change. One thing I didn't notice before is the semicolon at the end of the line. Would you mind removing it, as we're not using them in the rest of this example?

Such a small thing sorry!

Reply:
Old habits... Going to remove it.